src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+                jsonPath = fileOperationService.saveBinaryDataToSpecifiedWorkspace(
+                    invalidLineFileBinary,
+                    invalidLineFileFilenameProcessor.getInvalidLineFilename(),
+                    armDataManagementConfiguration.getTempBlobWorkspace(),
+                    appendUuidToWorkspace
+                );
+
+                if (jsonPath.toFile().exists()) {
+                    logResponseFileContents(jsonPath);
+                    ArmResponseInvalidLineRecord armResponseInvalidLineRecord = objectMapper.readValue(jsonPath.toFile(), ArmResponseInvalidLineRecord.class);
+                    UploadNewFileRecord uploadNewFileRecord = readInputJson(armResponseInvalidLineRecord.getInput());
+                    if (nonNull(uploadNewFileRecord)) {
+                        armBatchResponses.addResponseBatchData(Integer.valueOf(uploadNewFileRecord.getRelationId()),
+                                                               armResponseInvalidLineRecord, invalidLineFileFilenameProcessor);
+                    }
+                } else {
+                    log.warn("Failed to write invalid line file to temp workspace {}", invalidLineFileFilenameProcessor.getInvalidLineFileFilenameAndPath());
+                }
+            } catch (IOException e) {
+                log.error("Unable to write invalid line file to temporary workspace {} - {}",
+                          invalidLineFileFilenameProcessor.getInvalidLineFileFilenameAndPath(), e.getMessage());
+
+            } catch (Exception e) {
+                log.error("Unable to process ARM response invalid line file {}",
+                          invalidLineFileFilenameProcessor.getInvalidLineFileFilenameAndPath(), e);
+
+            } finally {
+                cleanupTemporaryJsonFile(jsonPath);
+            }
+        } else {
+            log.error("Unable to read ARM response invalid line file {}", invalidLineFileFilenameProcessor.getInvalidLineFileFilenameAndPath());
+        }
+    }
+
+    private void tryProcessInvalidLineFile(int externalObjectDirectoryId, InvalidLineFileFilenameProcessor invalidLineFileFilenameProcessor,
+                                           ArmResponseInvalidLineRecord armResponseInvalidLineRecord,
+                                           CreateRecordFilenameProcessor createRecordFilenameProcessor,
+                                           UploadFileFilenameProcessor uploadFileFilenameProcessor) {
+        try {
+            ExternalObjectDirectoryEntity externalObjectDirectory = getExternalObjectDirectoryEntity(externalObjectDirectoryId);
+            if (nonNull(externalObjectDirectory)) {
+                if (nonNull(armResponseInvalidLineRecord)) {
+                    //If the filename contains 0
+                    if (ARM_RESPONSE_INVALID_STATUS_CODE.equals(invalidLineFileFilenameProcessor.getStatus())) {
+
+                        processInvalidLineFile(armResponseInvalidLineRecord, externalObjectDirectory);
```

Comment:
author note: refactoring, moved existing logic into the method `processInvalidLineFile`